hypothesis-python/tests/cover/test_regex.py

```diff
@@ -78,8 +78,8 @@ def _test_matching_pattern(pattern, *, isvalidchar, is_unicode=False):
             assert r.search(c), msg % (pattern, c, c, unicodedata.category(c))
         else:
             assert not r.search(c), (
-                '"%s" supposed not to match "%s" (%r, category "%s"), '
-                "but it does" % (pattern, c, c, unicodedata.category(c))
+                f'"{pattern}" supposed not to match "{c}" ({c!r}, category '
+                f'{unicodedata.category(c)}"), but it does'
```

Comment:
Oh neat!

- we should use the `!r` specifier instead of manually inserting quotes; the current message will be suboptimal if the problem string includes `"`.
- I think the autofixer has dropped a double-quote after `category`!  We should confirm this and open an issue.

Reply:
oops, the double-quote drop was me, when adjusting the string length to fit under 88 chars (not automatically done by pyupgrade/shed/black!)